libcloudforensics/providers/kubernetes/netpol.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Kubernetes classes for wrapping NetworkPolicy APIs."""
+import abc
+import random
+import string
+from typing import Dict
+
+from kubernetes import client
+
+from libcloudforensics.providers.kubernetes import base
+
+
+class K8sNetworkPolicy(base.K8sNamespacedResource):
+  """Class representing a Kubernetes NetworkPolicy, enabling API calls."""
+
+  def Delete(self, cascade: bool = True) -> None:
+    """Override of abstract method. The cascade parameter is ignored."""
+    api = self._Api(client.NetworkingV1Api)
+    api.delete_namespaced_network_policy(self.name, self.namespace)
```

Comment:
On all `api` calls: do we need to wrap them in case they fail?

Reply:
Good question... I don't know what's best. These functions are not supposed to fail, as to create these objects you'd have to go through a cluster object, which does a permissions check when created. Maybe if we don't see a reason why they would fail, we should leave them unwrapped? What do you think?

Reply:
> These functions are not supposed to fail

Until they do! There can be many reasons why these would fail:

- Function called with wrong arguments
- Network interruption
- Race conditions (e.g. a resource is being modified by another program right before a call, which results in an error)
- ...

I'd opt to create dedicated error classes in `errors.py`, wrap these calls and raise an appropriate error. @sa3eed3ed wdyt?

Reply:
Okay I'd see a few options here, I'll list them below:
- Wrap all Kubernetes API calls with a try/except block (this may be repetitive because it will always be the same code, and one may easily forget to wrap it)
- Nest the Kubernetes API calls in a wrapping API class, creating methods that wrap their corresponding API call (advantage is now that this class will replace the regular Kubernetes API objects, so only wrapped calls will be exposed, but every Kubernetes API call that we need will have to be "redefined")
- Do Kubernetes API calls via a lambda which then gets wrapped and called (would could also do this via a class to enforce calls to be done this way)
- Alias the Kubernetes error in `errors.py` (this may be bad practice, and also it would not be possible to extend `LCFError`)
- Leave as is

Reply:
Hum... Yeah maybe we're over-engineering things here, after my 3rd review pass I saw just how many such calls there are and this might clutter the codebase too much / make it unnecessarily more complex. Let's leave it as is; if we get issues about this / folks complaining then we can re-think what to do.